src/sentry/utils/snuba.py

```diff
@@ -382,34 +387,39 @@ def transform_aliases_and_query(**kwargs):
     translated_columns = {}
     derived_columns = set()
 
-    selected_columns = kwargs['selected_columns']
-    groupby = kwargs['groupby']
-    aggregations = kwargs['aggregations']
-    conditions = kwargs['conditions'] or []
+    selected_columns = kwargs.get('selected_columns')
+    groupby = kwargs.get('groupby')
+    aggregations = kwargs.get('aggregations')
+    conditions = kwargs.get('conditions')
     filter_keys = kwargs['filter_keys']
+    arrayjoin = kwargs.get('arrayjoin')
+    rollup = kwargs.get('rollup')
+    orderby = kwargs.get('orderby')
+
+    if selected_columns:
+        for (idx, col) in enumerate(selected_columns):
+            if isinstance(col, list):
+                # if list, means there are potentially nested functions and need to
+                # iterate and translate potential columns
+                parse_columns_in_functions(col)
+                selected_columns[idx] = col
```

Comment:
I forget how python works internally, but I wondered if you get any issues mutating `selected_columns` while iterating on them? I think that's a plain python dict, and my fear is that while you might not get an error, the `enumerate` method may still skip over keys in the dictionary depending upon what is added. 

Have you read something in the documentation that guarantees the ordering of the keys so that you never skip over one? I'm not aware that it works like that for python 2 dicts. I'd recommend restructuring this to not mutate and iterate on the dictionary at the same time. 

Because I can see that this has been done in other areas of this code, @markstory or @wedamija love your thoughts here too.

Reply:
I think this is fine, because `selected_columns` is a list.

Reply:
`selected_columns` is a list of strings, not a dict